code/modules/vending/clothingbooth.dm

```diff
@@ -100,14 +100,19 @@ var/list/clothingbooth_items = list()
 				usr << output("preview_overlay.png", "ClothingBooth.browser:preview")
 
 	Click()
-		if((usr in src.contents) && (src.open == 0))
+		if(!ishuman(usr))
+			boutput(usr,"<span style=\"color:red\">Human clothes don't fit you, silly!</span>")
+			return
+		if((usr in src) && (src.open == 0))
 			if(istype(usr.equipped(),/obj/item/spacecash))
 				var/obj/item/dummycredits = usr.equipped()
 				src.money += dummycredits.amount
 				dummycredits.amount = 0
 				qdel(dummycredits)
+				return
 			else
 				uisetup(usr)
+				return
 		..()
```

Comment:
Won't not calling parent cause problems with stuff? Like not being able to... paint the booth with a paint can for example, or not being able to deconstruct it?

Reply:
i will also look into this. (sorry debugging my first patch is **AAAAAAAAAAAA**)

Reply:
This should be fine as is. Though others can change it later if needed.